docs/reference/ingest.asciidoc

```diff
@@ -266,6 +266,128 @@ Use the <<index-final-pipeline,`index.final_pipeline`>> index setting to set a
 final pipeline. {es} applies this pipeline after the request or default
 pipeline, even if neither is specified.
 
+[discrete]
+[[pipelines-for-fleet-elastic-agent]]
+=== Pipelines for {fleet} and {agent}
+
+{fleet-guide}/index.html[{fleet}] automatically adds ingest pipelines for its
+integrations. {fleet} applies these pipelines using <<index-templates,index
+templates>> that include <<set-default-pipeline,pipeline index settings>>. {es}
+matches these templates to your {fleet} data streams based on the
+{fleet-guide}/data-streams.html#data-streams-naming-scheme[stream's naming
+scheme].
+
+WARNING: Do not change {fleet}'s ingest pipelines or use custom pipelines for
+your {fleet} integrations. Doing so can break your {fleet} data streams.
+
+{fleet} doesn't provide an ingest pipeline for the **Custom logs** integration.
+You can safely specify a pipeline for this integration in one of two ways:
+
+**Option 1: Index template**
+
+. Create an <<index-templates,index template>> that includes
+<<set-default-pipeline,pipeline index settings>>. You can create this template
+using {kib}'s <<manage-index-templates,**Index Management**>> feature or the
+<<indices-put-template,create index template API>>. Ensure the index template is
+<<create-a-data-stream-template,data stream enabled>>.
++
+--
+To avoid naming collisions, the template's index pattern should match
+`logs-<dataset-name>-*`.
+
+For example, the following request creates a template matching `logs-my_app-*`.
+The template uses a component template that contains the
+`index.default_pipeline` index setting.
+
+[source,console]
+----
+# Creates a component template for index settings
+PUT _component_template/my-settings-component-template
+{
+  "template": {
+    "settings": {
+      "index.default_pipeline": "my-pipeline"
```

Comment:
I would have in our recommendation to name the pipeline the same as the data stream to just make it a habit.

Reply:
Thanks @ruflin. I incorporated this naming convention into the docs with 5d584dc549799dc4a36be76debb578c30a6d2b4d and aeca6c3572fbc84ba8fa1eb2bd214ffcd62df0bf.